staging/src/k8s.io/kube-aggregator/pkg/apiserver/apiserver.go

```diff
@@ -343,6 +343,40 @@ func (c completedConfig) NewWithDelegate(delegationTarget genericapiserver.Deleg
 		return nil
 	})
 
+	if utilfeature.DefaultFeatureGate.Enabled(genericfeatures.AggregatedDiscoveryEndpoint) {
+		s.discoveryAggregationController = NewDiscoveryManager(
+			// Use aggregator as the source name to avoid overwriting native/CRD
+			// groups
+			s.GenericAPIServer.AggregatedDiscoveryGroupManager.WithSource(aggregated.AggregatorSource),
+		)
+
+		// Setup discovery endpoint
+		s.GenericAPIServer.AddPostStartHookOrDie("apiservice-discovery-controller", func(context genericapiserver.PostStartHookContext) error {
+			// Discovery aggregation depends on the apiservice registration controller
+			// having the full list of APIServices already synced
+			select {
+			case <-context.StopCh:
+			// Context cancelled, should abort/clean goroutines
+			case <-apiServiceRegistrationControllerInitiated:
```

Comment:
noting for future-self that this means we have called `setInfoForAPIService` before this is closed, so every `discoveryManager.apiServices` that is known (lister hasSync is checked) is added and thus the `markInitialStale` will mark the starting list as stale.

Reply:
With the new change, every apiservice will be processed and in discovery (either stale if unreachable or has all gvs)